cron.ts

```diff
@@ -1,77 +1,103 @@
 import { IsNull, Not } from "typeorm";
 import config from "./config";
-import { DiscordMemberRepository, DiscordServerRepository } from "./db";
+import { DiscordMemberRepository, DiscordServerConfigRepository } from "./db";
 import { DiscordMember } from "./db/entity/DiscordMember";
-import { DiscordServer } from "./db/entity/DiscordServer";
+import { DiscordServerConfig } from "./db/entity/DiscordServerConfig";
 import { restDiscordClient } from "./discord/client";
 import { addRole, removeRole } from "./discord/role";
 import modules from "./starkyModules";
 import { StarkyModule } from "./starkyModules/types";
 
-const refreshDiscordServers = async () => {
-  const discordServers = await DiscordServerRepository.find();
-  for (let discordServer of discordServers) {
-    await refreshDiscordServer(discordServer);
+const refreshDiscordServerConfigs = async () => {
+  const DiscordServerConfigs = await DiscordServerConfigRepository.find();
+  for (let DiscordServerConfig of DiscordServerConfigs) {
+    await refreshDiscordServerConfig(DiscordServerConfig);
   }
 };
 
-export const refreshDiscordServer = async (discordServer: DiscordServer) => {
-  console.log(`[Cron] Refreshing discord server ${discordServer.id}`);
+export const memberStarknetAddress = async (discordMember: DiscordMember) => {
+  const discordMemberWithAddress = await DiscordMemberRepository.findOneBy({
+    starknetWalletAddress: Not(IsNull()),
+    discordMemberId: discordMember.discordMemberId,
+    DiscordServerId: discordMember.DiscordServerId,
+  });
+  if (discordMemberWithAddress == null) return;
+  discordMember.starknetWalletAddress =
+    discordMemberWithAddress.starknetWalletAddress;
+  //console.log(discordMember);
+  return;
+};
+
+export const refreshDiscordServerConfig = async (
+  DiscordServerConfig: DiscordServerConfig
+) => {
+  console.log(
+    `[Cron] Refreshing discord server ${DiscordServerConfig.DiscordServerId}, configuration ${DiscordServerConfig.id}`
+  );
   const discordMembers = await DiscordMemberRepository.find({
     where: {
-      discordServer,
-      starknetWalletAddress: Not(IsNull()),
+      DiscordServerConfig,
```

Comment:
same question : why not keep the discord server model to still do the same query here, but loop on each configuration to find what role(s) to apply?